lib/vsc/mympirun/mpi/openmpi.py

```diff
@@ -85,16 +99,26 @@ def set_mpiexec_options(self):
         """Set mpiexec options"""
         super(OpenMPI, self).set_mpiexec_options()
 
-        if self.options.hybrid:
-            # specify number of processes to start per node if --hybrid is used
-            procs_per_node = self.multiplier * self.options.hybrid
-            self.mpiexec_options.add(['--map-by', 'ppr:%s:node' % procs_per_node])
+        if self.is_oversubscribed():
+            logging.debug("Allow oversubscription")
+            over = ''
         else:
-            # map MPI processes by core (default is --map-by numa),
-            # and bind to core (default is --bind-to numa' when # ranks > 2),
-            # to match default behaviour of Intel MPI;
-            # this is important for performance for OpenFOAM for example, especially on AMD Rome CPUs
-            self.mpiexec_options.add(['--map-by', 'core', '--bind-to', 'core'])
+            over = 'NO'
+
+        # make sure we start enough per node so it can fill the total_number_of_processes
+        tnop = self.total_number_of_processes()
+        un = len(self.nodes_uniq)
+        ppn = int(math.ceil(tnop / un))
+        logging.debug("Setting up map for %s (%s total number of processes on %s unique nodes)", ppn, tnop, un)
+
+        mapby = [
+            'ppr', "%s" % ppn, 'node',
```

Comment:
Can we add some comments here to explain what each of these lines mean?
Maybe also include a pointer to https://www.open-mpi.org/doc/current/man1/mpirun.1.php on top of `mapby =`?

`ppr` vs `ppn` is very confusing here, but this line basically is:

```
# specify number of cores to start per node;
# 'ppr' is processes per resource type, with 'node' being the resource type
```

Also, use `str(cores_per_node)` rather than `"%s" % cores_per_node`?

Reply:
i'm not repeating text from a manpage here.